java/src/main/java/ai/rapids/cudf/ColumnVector.java

```diff
@@ -257,6 +348,58 @@ public boolean hasNulls() {
   // DATA MOVEMENT
   /////////////////////////////////////////////////////////////////////////////
 
+  private final HostColumnVector.NestedHostColumnVector copyToHostNestedHelper(ColumnViewPointerAccess deviceCvPointer) {
+    // Now some of this we could speed up with caching, but conceptually it should be like the following...
+    // None of these need to be closed because they do not own any data, just views conceptually.
+    if (deviceCvPointer == null) {
+      return null;
+    }
+    HostMemoryBuffer hostOffsets = null;
+    HostMemoryBuffer hostValid = null;
+    HostMemoryBuffer hostData = null;
+    List<HostColumnVector.NestedHostColumnVector> children = new ArrayList<>();
+    MemoryBuffer currData = null;
+    MemoryBuffer currOffsets = null;
+    MemoryBuffer currValidity = null;
+    try {
+
+      long currRows = deviceCvPointer.getNumRows();
+      DType currType = deviceCvPointer.getDataType();
+      currData = deviceCvPointer.getDataBuffer();
+      currOffsets = deviceCvPointer.getOffsetBuffer();
+      currValidity = deviceCvPointer.getValidityBuffer();
+      if (currData != null) {
+        hostData = HostMemoryBuffer.allocate(currData.length);
+        hostData.copyFromDeviceBuffer((DeviceMemoryBufferView)currData);
```

Comment:
nit: Can we have a follow on issue to remove the casting?  We should be able to use a sub-interface or sub-class of `ColumnViewPointerAccess` that returns `DeviceMemoryBufferView` instead.  That way we can guarantee that the cast will work by way of the compiler.  Don't need to do it now we are in burn-down and this is a private method, so I would much rather just get this in as is.

Reply:
Templatized to make the casting go away.